packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/gollm/GoLLMController.java

```diff
@@ -858,32 +858,57 @@ public ResponseEntity<TaskResponse> createEnrichModelMetadataTask(
 		// Update State Grounding
 		if (!model.isRegnet()) {
 			final List<State> states = model.getStates();
-			states.forEach(state -> TaskUtilities.performDKGSearchAndSetGrounding(miraProxy, state));
+			states.forEach(state -> {
```

Comment:
I'm not sure if these are going to work as expected? if the element in these lists are null, will `state = new State();` correctly init it in the list of objects? I think that is effectively the same as doing a `continue` because its not added back into the lists?

Reply:
why not just filtering first?
```java
states.stream()
	.filter(Objects::nonNull)   // filters out null values
	.forEach(state -> TaskUtilities.performDKGSearchAndSetGrounding(miraProxy, state));
```

Reply:
we need to make sure that there is a base grounding since the dkg search can return a null value. This null value causes Mira to fail.